### PR TITLE
Add support for gRPC reflection

### PR DESCRIPTION
**Pull request by aseovic**

### CODE REVIEW

File: grpc/server/src/main/java/io/helidon/grpc/server/BindableServiceImpl.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (tomas-langer) [inline grpc/server/src/main/java/io/helidon/grpc/server/BindableServiceImpl.java:2]: This is wrong copyright format. Please use `2019, 2021`.

File: grpc/server/src/main/java/io/helidon/grpc/server/GrpcServerImpl.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2021, Oracle and/or its affiliates.
```
Comment (tomas-langer) [inline grpc/server/src/main/java/io/helidon/grpc/server/GrpcServerImpl.java:2]: Wrong copyright format

File: grpc/server/src/main/java/io/helidon/grpc/server/ServiceDescriptor.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (tomas-langer) [inline grpc/server/src/main/java/io/helidon/grpc/server/ServiceDescriptor.java:2]: Wrong copyright format.

File: grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+        return new ProtoReflectionService();
+    }
+
+    /**
+     * Retrieves the index for services of the server that dispatches the current call. Computes
+     * one if not exist. The index is updated if any changes to the server's mutable services are
+     * detected. A change is any addition or removal in the set of file descriptors attached to the
+     * mutable services or a change in the service names.
+     */
+    private ServerReflectionIndex getRefreshedIndex() {
```
Comment (tomas-langer) [inline grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java:94]: How often is this called? Hard synchronization is expensive... 
Do we need to do it as pull approach (e.g. do we need to query the server for all services, as opposed to getting events about mutable services being changed)?

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java (hunk 4 above)
Comment (aseovic) [reply grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java:94]: In reality, only once, so it shouldn't be a big deal.

Services don't really change at runtime, so there shouldn't be any changes that need refreshing.

**Comment by aseovic** [comment]
@tomas-langer Any chance we can get this merged?

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**